docs/developers/developer-reference/sc-contract-calls.md

```diff
@@ -122,31 +417,61 @@ fn my_endpoint_callback(
 }
 ```
 
-To assign this callback to the aforementioned async call, we hook it like this:
+The `#[call_result]` argument interprets the output of the called endpoint and must almost always be of type `ManagedAsyncCallResult`. This type decodes the error status from the VM, more about it [here](/developers/data/multi-values#standard-multi-values). Its type argument must match the return type of the called endpoint.
+
+To assign this callback to the aforementioned async call, we hook it after `async_call`, but before `call_and_exit`:
+
 ```rust
 #[endpoint]
 fn caller_endpoint(&self) {
-	// other code here
+	// previous code here
 
-	self.contract_proxy(callee_sc_address)
-		.my_endpoint(my_biguint_arg)
+	self.callee_contract_proxy(callee_sc_address)
+		.callee_endpoint(my_biguint_arg)
 		.async_call()
-		.with_callback(self.callbacks().my_endpoint_callback())
+		.with_callback(self.callbacks().callee_endpoint_callback())
```

Comment:
`callee_endpoint_callback` is called here, should it rather be a function pointer? Without the parantheses?

Reply:
I wish we had the power to redefine function pointers. We don't.

The next best thing: fooling the developer. This is no the actual callback, it is the callback proxy, that function actually produces a `CallbackCall`.

I won't be writing about it today. Maybe another day.

Marvel in the mischief!